File: tests/benchmarks/resolution/fixtures/clojure/repository.clj
(ns app.repository)

(defn new-repo []
  (atom {}))

(defn save [repo name email]
  (swap! repo assoc name {:name name :email email})
  :ok)

(defn find-by-id [repo id]
  (get @repo id))

(defn delete [repo id]
  (if (contains? @repo id)
    (do (swap! repo dissoc id) :ok)
    {:error "not found"}))

(defn count [repo]
  (count-entries @repo))

(defn- count-entries [data]
  (clojure.core/count data))

File: tests/benchmarks/resolution/fixtures/julia/repository.jl
module Repository

function new_repo()
    Dict{String, Tuple{String, String}}()
end

function save(repo, name, email)
    repo[name] = (name, email)
    :ok
end

function find_by_id(repo, id)
    if haskey(repo, id)
        repo[id]
    else
        error("not found")
    end
end

function delete(repo, id)
    if haskey(repo, id)
        delete!(repo, id)
        :ok
    else
        error("not found")
    end
end

function count(repo)
    count_entries(repo)
end

function count_entries(repo)
    length(repo)
end

end # module Repository
